src/main/kotlin/pl/put/smartgarden/infra/api/UserController.kt

```diff
@@ -119,4 +129,130 @@ class UserController(val userService: UserService) {
+        @RequestHeader("Authorization") token: String,
+        @RequestBody locationDto: LocationDto
+    ): UserResourceDto =
+        userService.setLocation(token, locationDto)
+
+    @PutMapping("/areas/{areaId}/next-irrigation")
+    @ApiOperation("Set next irrigation time for selected area.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 403, message = "Unactivated")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun setNextIrrigationTime(
+        @RequestHeader("Authorization") token: String,
+        @RequestBody irrigationTimeDto: IrrigationTimeDto,
+        @PathVariable("areaId") areaId: String
+    ): NextIrrigationDto =
+        userService.setNextIrrigationTime(token, areaId, irrigationTimeDto)
+
+    @PostMapping("/areas/{areaId}/irrigate-now")
+    @ApiOperation("Irrigate selected area.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 403, message = "Unactivated")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun irrigateArea(
+        @RequestHeader("Authorization") token: String,
+        @PathVariable("areaId") areaId: String
+    ): Unit =
+        userService.irrigateArea(token, areaId)
+
+    @PutMapping("/areas/{areaId}/link-sensor/{sensorId}")
+    @ApiOperation("Add selected sensor to selected area.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 403, message = "Unactivated")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun linkSensorToArea(
+        @RequestHeader("Authorization") token: String,
+        @PathVariable("areaId") areaId: String,
+        @PathVariable("sensorId") sensorId: String
+    ): List<AreaDto> =
+        userService.linkSensorToArea(token, areaId, sensorId)
+
+    @PutMapping("/areas/unlink-sensor/{sensorId}")
+    @ApiOperation("Remove selected sensor from its area.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 403, message = "Unactivated")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun unlinkSensorFromArea(
+        @RequestHeader("Authorization") token: String,
+        @PathVariable("sensorId") sensorId: String
+    ): List<AreaDto> =
+        userService.unlinkSensorFromArea(token, sensorId)
+
+    @GetMapping("/sensors")
+    @ApiOperation("Get not linked sensors.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 403, message = "Unactivated")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun getNotLinkedSensors(@RequestHeader("Authorization") token: String): List<Sensor> =
```

Comment:
Osobiście jestem przeciwnikiem zwracania encji bazodanowych w odpowiedzi. Zdaję sobię sprawę że w naszym projekcie może to nie mieć znaczenia, ale polecam poczytać artykuł: https://thorben-janssen.com/dont-expose-entities-in-api/